src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -389,12 +381,12 @@ public void setName(String name) {
      * @return interned string
      */
     protected String getCachedString(String str) {
-        String internedString = stringCache.get(str);
+        WeakReference<String> internedString = stringCache.get(str);
```

Comment:
I thought you need to also check that the referenced string inside of the WeakReference hasn't been GC'ed...shouldn't the if statement be something like
```
if ( internedString == null || internedString.get()==null) 
```
???